miden-lib/asm/kernels/transaction/lib/prologue.masm

```diff
@@ -481,6 +427,37 @@ proc.process_account_data
     movup.4 drop
     # => [ACCT_HASH]
 
+    # assert the account id matches the account id in global inputs
+    exec.memory::get_global_acct_id
+    exec.memory::get_acct_id
+    assert_eq.err=ERR_PROLOGUE_ACCT_ID_MISMATCH
+    # => [ACCT_HASH]
+
+    # store a copy of the initial nonce in global inputs
+    exec.memory::get_acct_nonce
+    exec.memory::set_init_nonce
+    # => [ACCT_HASH]
+
+    # set the new account code commitment to the initial account code root this is used for managing
+    # code commitment updates
+    exec.memory::get_acct_code_commitment
+    exec.memory::set_new_acct_code_commitment
+    # => [ACCOUNT_CODE_COMMITMENT, ACCT_HASH]
+
+    # validates and stores account procedures in memory.
+    exec.validate_account_procedures
+    # => [ACCT_HASH]
+
+    # validates and stores account storage slots in memory.
+    exec.memory::get_acct_storage_root
+    exec.validate_account_storage_slots
+    # => [ACCT_HASH]
```

Comment:
I would move this above processing account code. This way, by the time we process account code we already know how many slots are in the storage. This will allow us to do additional validation in the future.